### DISCUSSION TITLE
Is there a way to not call `useQuery` directly?

### DISCUSSION THREAD

**Discussion by TheDutchCoder** [Akryum/vue-apollo#1321] [Q&A] [answered]
With `useMutation` you can use `const { mutation } = useMutation()` and call `mutation()` later on to actually run the code.

This doesn't seem to be the case with `useQuery`, which just immediately runs the query, but wouldn't it be logical to define the query and then call it at a later point in time?

Or am I missing something?

I just want to be able to do something like `const { query: getThings } = useQuery()` and call `getThings` when a user clicks a button for example.

**Comment by Akryum** [comment, marked as answer]
You can use `useLazyQuery`. Probably needs more documentation on this.

**Reply by TheDutchCoder** [reply]
Interesting, I was indeed not aware of that, I'll give that a shot, thanks!

**Reply by TheDutchCoder** [reply]
Yes perfect! That's exactly what I needed!
